src/local.rs

```diff
@@ -72,6 +71,17 @@ pub(crate) enum Error {
         path: String,
         source: io::Error,
     },
+
+    #[snafu(display("Path \"{}\" contained non-unicode path segment", path.display()))]
+    NonUnicode {
+        path: std::path::PathBuf,
+    },
+
+    #[snafu(display("Path \"{}\" contained bad path segment: \"{}\"", path.display(), segment))]
```

Comment:
Can we please update this to explain what was "bad" about the segment? Perhaps by embedding the BadSegment error?